### ISSUE TITLE
False positive "Unused variable" report

### ISSUE THREAD

**Issue by Strate** [palantir/tslint#875]
Hi all! I found a case, when false-positive "Unused variable" reports. It comes alongside with JSX:

```ts
import * as ReactIntl from "react-intl"

export function render() {
    return <ReactIntl.FormattedMessage />
}
```

fires me "Unused variable ReactIntl"

**Comment by adidahiya** [comment]
@Strate which version of TSLint are you using?

**Comment by Strate** [comment]
```
└─┬ tslint-loader@1.0.2
  └── tslint@2.5.1
```

**Comment by adidahiya** [comment]
Try updating both TS and TSLint to the latest versions. Is that feasible for your project?

**Comment by Strate** [comment]
```
typescript@1.7.5
tslint@3.2.0
```
Same issue

**Comment by alexgorbatchev** [comment]
```
typescript@1.7.5
tslint@3.2.2
```

seeing

```
ERROR in ./src/reducers/index.ts
[1, 8]: unused variable: 'counter'
```

when `index.ts` is

```javascript
import counter from './counter';

export default { counter };
```

**Comment by JKillian** [comment]
Hi @alexgorbatchev, check out #684 which was the same issue as yours. It should be fixed now if you update to `typescript@1.8.0-dev.20160116` or newer

**Comment by bluekeyes** [comment]
I'm also seeing the original issue with tslint 3.3.0 and typescript 1.7.5, which is making it annoying to "namespace" related components by exporting them from an index module.

**Comment by adidahiya** [comment]
I think this is another variant of the issue in #325, which is due to a compiler bug. I'll ping the TS team to see if there are any updates.

**Comment by JKillian** [comment]
This should be fixed in the newest versions of TypeScript, see https://github.com/palantir/tslint/issues/1010 for more info
